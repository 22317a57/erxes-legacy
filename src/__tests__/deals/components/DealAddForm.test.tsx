// test passed
import { mount, shallow } from 'enzyme';
import DealAddForm from 'modules/deals/components/DealAddForm';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('DealAddForm component', () => {
  const defaultProps = {
    add: (name: string, callback: () => void) => null,
    closeModal: () => null
  };

  test('renders shallow successfully', () => {
    shallow(<DealAddForm {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<DealAddForm {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<DealAddForm {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
