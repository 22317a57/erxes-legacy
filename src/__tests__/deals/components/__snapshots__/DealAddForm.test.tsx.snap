// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealAddForm component snapshot matches 1`] = `
<form
  className="sc-fihHvN evWDap"
  onSubmit={[Function]}
>
  <div
    className="sc-RbTVP bSyGnE sc-jdfcpN hYPvAm"
  >
    <div
      className="sc-hMrMfs hcFoZm"
    >
      <label
        className="sc-ebFjAB bHuvQk"
      >
        Name
        <span>
           *
        </span>
      </label>
      <input
        autoFocus={true}
        className="sc-kaNhvL bKDVoH"
        disabled={false}
        onChange={[Function]}
        required={false}
      />
    </div>
  </div>
  <div
    className="sc-jxGEyO dAqHNM"
  >
    <button
      className="sc-ifAKCX iSmYrG"
      icon="cancel-1"
      onClick={[Function]}
      type="button"
    >
      <i
        className="icon-cancel-1 sc-bwzfXH hpxbGw"
        icon="cancel-1"
      />
      <span>
        Close
      </span>
    </button>
    <button
      className="sc-ifAKCX cQdFDO"
      disabled={false}
      icon="checked-1"
      type="submit"
    >
      <i
        className="icon-checked-1 sc-bwzfXH hpxbGw"
        icon="checked-1"
      />
      <span>
        Save
      </span>
    </button>
  </div>
</form>
`;
